rust/lance/src/dataset/builder.rs

```diff
@@ -271,6 +271,7 @@ impl DatasetBuilder {
 
     #[instrument(skip_all)]
     pub async fn load(mut self) -> Result<Dataset> {
```

Comment:
The codes like this `ds = lance.dataset("/tmp/test.lance")` will not trigger `opening` event. It would trigger this load method(both `write` and `read` paths would trigger this method). So it would be better to add a `loading` event.

Reply:
The original intention is to track only the `read` operation. I misunderstood that `open` can do this. The `load` method is called in both `read` and `write` operations. 

Finally, I found that the `loading` event traces in the `load` method can also be accepted.

Now my thought is to introduce `loading` in the `load` method, and remove `opening`(if a dataset is first written, `opening` may be confused). WDYT?

Reply:
> The original intention is to track only the read operation

if that is the case, I think we should add tracing event in the scanner

> The load method is called in both read and write operations.

`Dataset::open` also eventually calls `DatasetBuilder::load`, so I think we can remove the one in `Dataset::open`. At that point, it seems like it's just a matter of calling it `open` or `load`, I am fine with both.

Reply:
> so I think we can remove the one in Dataset::open

Yes.

> At that point, it seems like it's just a matter of calling it open or load, I am fine with both.

I prefer `load`.